CharlieBackend.Business/Services/HomeworkService.cs

```diff
@@ -127,14 +126,9 @@ public async Task<Result<HomeworkDto>> UpdateHomeworkAsync(long homeworkId, Home
             foundHomework.DueDate = updateHomeworkDto.DueDate;
             foundHomework.TaskText = updateHomeworkDto.TaskText;
 
-            var mentor = await _unitOfWork.MentorRepository.GetByIdAsync(foundHomework.MentorId);
-            foundHomework.MentorId = updateHomeworkDto.MentorId;
-            foundHomework.Mentor = mentor;
-
-            var studentGroup = await _unitOfWork.StudentGroupRepository
-                    .GetByIdAsync(updateHomeworkDto.StudentGroupId);
-            foundHomework.StudentGroupId = updateHomeworkDto.StudentGroupId;
-            foundHomework.StudentGroup = studentGroup;
+            var lesson = await _unitOfWork.LessonRepository.GetByIdAsync(foundHomework.LessonId);
+            foundHomework.LessonId = updateHomeworkDto.LessonId;
```

Comment:
I don't think that we have to update the lesson id. Usually, homework is bonded to a specific lesson, and we want to update only text, attachments and so on.